datasets/plugins/batch/hive_dataset.py

```diff
@@ -203,8 +212,34 @@ def write_spark(
                 .saveAsTable(self.hive_table)
             )
 
+            create_view_query = f"""
+            CREATE OR REPLACE VIEW {self.hive_table}_latest AS
+            SELECT * FROM (
+                SELECT s.*, RANK() OVER(ORDER BY run_time DESC) rank FROM {self.hive_table} s
+            ) s
+            WHERE rank=1
+            """
+            spark.sql(create_view_query)
+
     def __repr__(self):
         return (
             f"HiveDataset({self.hive_table=},{self.name=},{self.key=},{self.partition_by=},"
             f"{self.run_id=},{self.run_time=},{self.columns=},{self.mode=}"
         )
+
+
+def _retry_with_backoff(func: Callable, retries=5, backoff_in_seconds=1):
+    i = 0
+    while True:
+        try:
+            return func()
+        except Exception as e:
+            if i >= retries:
+                print(f"Failed after {retries} retries:")
+                raise
+            else:
+                i += 1
+                sleep = backoff_in_seconds * 2 ** i + random.uniform(0, 5)
```

Comment:
this is to wait for tmp to be cleaned up?

Reply:
done!